client/gcp/gcp.go

```diff
@@ -0,0 +1,212 @@
+
+type unitPrice struct {
+	CurrencyCode string  `json:"currencyCode"`
+	Unit         string  `json:"units"`
+	Nano         float64 `json:"nanos"`
+}
+
+type tieredRate struct {
+	UnitPrice unitPrice `json:"unitPrice"`
+}
+
+type pricingExpression struct {
+	UsageUnit      string        `json:"usageUnit"`
+	TieredRateList []*tieredRate `json:"tieredRates"`
+}
+
+type pricingInfo struct {
+	PricingExpression pricingExpression `json:"pricingExpression"`
+}
+
+type category struct {
+	ServiceDisplayName string `json:"serviceDisplayName"`
+	ResourceFamily     string `json:"resourceFamily"`
+	ResourceGroup      string `json:"resourceGroup"`
+	UsageType          string `json:"usageType"`
+}
+
+type offer struct {
+	ID                string        `json:"skuId"`
+	Description       string        `json:"description"`
+	Category          category      `json:"category"`
+	PricingInfo       []pricingInfo `json:"pricingInfo"`
+	ServiceRegionList []string      `json:"serviceRegions"`
+}
+
+type pricing struct {
+	OfferList     []*offer `json:"skus"`
+	NextPageToken string   `json:"nextPageToken"`
+}
+
+func httpHelper(nextpageToken string) ([]*offer, error) {
+	endpoint := PriceInfoEndpoint
+	if nextpageToken != "" {
+		endpoint = fmt.Sprintf("%v&pageToken=%v", PriceInfoEndpoint, nextpageToken)
+	}
+
+	req, err := http.NewRequest("GET", endpoint, nil)
+	req.Header.Add("User-Agaent", `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15`)
+	req.Header.Add("Connection", `keep-alive`)
+	req.Header.Add("Accept", `text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8`)
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Fail to fetch the info file, [internal]: %v", err)
+	}
+	if res.StatusCode != 200 {
+		return nil, fmt.Errorf("An http error occur , [internal]: %v", err)
+	}
+
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Fail when reading response data , [internal]: %v", err)
+	}
+	var offerList []*offer
+	rawData := &pricing{}
+	if err := json.Unmarshal(data, rawData); err != nil {
+		return nil, fmt.Errorf("Fail when unmarshaling response data , [internal]: %v", err)
+	}
+	offerList = rawData.OfferList
+
+	if rawData.NextPageToken != "" {
+		nextPage, err := httpHelper(rawData.NextPageToken)
+		if err != nil {
+			return nil, err
+		}
+		offerList = append(offerList, nextPage...)
+		return offerList, nil
+	}
+
+	return offerList, nil
+}
+
+// GetOffer return the offers provide by GCP.
+func (c *Client) GetOffer() ([]*client.Offer, error) {
+	rawOffer, err := httpHelper("")
+	if err != nil {
+		return nil, err
+	}
+
+	var offerList []*client.Offer
+	incrID := 0
+	for _, rawOffer := range rawOffer {
+		// This condition will filter resource like Network.
+		if rawOffer.Category.ResourceFamily != "ApplicationServices" ||
```

Comment:
Shall we do the filtering by inclusive filter or exclusive filter (this)? If GCP may introduce other types of ResourceFamily later, we may want to do inclusive filtering here otherwise code will break in the future.